enclave/core/wchar.c

```diff
@@ -0,0 +1,12 @@
+// Copyright (c) Open Enclave SDK contributors.
```

Comment:
Do we need to add an implementation here, or can we reuse the 3rdparty/musl/musl/src/string/wcslen.c implementation under the `NEEDS_STDC_NAMES` section introduced in #2767?

Reply:
The mechanism in #2767 is for compiling internal files which are part of oecore but use stdc names. Since wchar.h will be included by code compiled by users, I would prefer to use the oe_* name.

Reply:
I want to minimize the use of NEEDS_STDC_NAMES as much as possible

Reply:
Side note: As you start making these distinctions in use cases, it would be useful to start documenting the guidance for this as well to contributors.

Reply:
Sounds good, I can do that as part of #2767